notebooks/demo_latticemaze.ipynb

```diff
@@ -72,9 +73,22 @@
             },
             "outputs": [],
             "source": [
-                "maze = LatticeMazeGenerators.gen_dfs(np.array([4,4]))\n",
+                "N: int = 10\n",
                 "\n",
-                "maze"
+                "maze = LatticeMazeGenerators.gen_dfs(np.array([N,N]))\n",
+                "tgt_maze: TargetedLatticeMaze = TargetedLatticeMaze.from_lattice_maze(maze, (0,0), (N-1,N-1))\n",
+                "solved_maze: SolvedMaze = SolvedMaze.from_targeted_lattice_maze(tgt_maze)\n",
```

Comment:
There's probably a good reason for this, but what is the benefit of having three different classes over of having one lattice_maze class with optional variables of start_pos, end_pos, solution_path?

Reply:
this is a really going point, and I considered doing it this way! I'm not 100% sure of this path, but my reasoning mostly boils down to:
- type safety: having to constantly check things like `if maze.start_pos is not None` seems like it would add a lot of code and also make it easy to shoot yourself in the foot. static type checkers will play nicer with having a separate class for this
- existing infrastructure: we already had the separation between `LatticeMaze` and `SolvedMaze`, which a lot of code depended on. replacing each usage of that would complicate things, doubly so if we had to perform the above checks constantly